### ISSUE TITLE
Bad credentials file breaks auth chain

### ISSUE THREAD

**Issue by weirdtales** [aws/aws-sdk-go-v2#220]
### Version of AWS SDK for Go?
v0.0.0-20180612204834-f401f4e72ca1

### Version of Go (`go version`)?
go version go1.11 linux/amd64

### What issue did you see?
On an EC2 instance, if `~/.aws/credentials` is a directory, `LoadDefaultAWSConfig()` fails.
This can happen due to `docker` mounting a non-existent credentials file.

**Comment by dedelala** [comment]
This can also occur if `~/.aws/config` is a directory in the same fashion.

**Comment by xibz** [comment]
Hello @weirdtales and @dedelala, thank you for reaching out to us. Can we get the error that is returned?

**Comment by dedelala** [comment]
```

**Comment by dedelala** [comment]
wait, my mistake, it's actually this one
```
failed to load shared config file, <home-dir>/.aws/config, open <home-dir>/.aws/config: permission denied
```

**Comment by xibz** [comment]
@dedelala - What version of the SDK are you using? If you try using HEAD, do you still see the same error?

**Comment by dedelala** [comment]
Have observed with a couple of versions, other than OP they are `v2.0.0-preview.4+incompatible` and `v0.0.0-20180511210858-aa81187f04fb`

**Comment by xibz** [comment]
Can you try master? There was a change with how we are parsing ini files.

**Comment by dedelala** [comment]
Confirmed unable to repro using HEAD

**Comment by xibz** [comment]
Awesome, thanks! We have a PR out to sync to V1. We can then create another release

**Comment by xibz** [comment]
We just released [Release v2.0.0-preview.5 (2018-09-27)](https://github.com/aws/aws-sdk-go-v2/releases/tag/v2.0.0-preview.5). If you guys have any other issues, please let us know!
